nipy/algorithms/statistics/formula/formulae.py

```diff
@@ -575,33 +585,47 @@ def __add__(self, other):
         """
         if not is_formula(other):
             raise ValueError('only Formula objects can be added to a Formula')
-        f = self.__class__(np.hstack([self.terms, other.terms]))
+        f = Formula(np.hstack([self.terms, other.terms]))
```

Comment:
so now even subclasses would return Formula.  I am not familiar much with this codebase, but is that desired? why in previous form it was faulty?